File: C++/majorityElement.cpp
class Solution {
    public:
        int majorityElement(vector<int>& nums) 
        {
            std::unordered_map<int, int> elemCountMap;
            for (const auto& num : nums) 
            {
                if (elemCountMap.find(num) == elemCountMap.end())
                {
                    elemCountMap[num] = 1;
                }
                else
                {
                    ++elemCountMap[num];
                }
            }
    
            int maxCount = 0;
            int elem = -1;
            for (const auto& elemCount : elemCountMap)
            {
                if (elemCount.second > nums.size() / 2) 
                {
                    if (elemCount.second > maxCount)
                    {
                        maxCount = elemCount.second;
                        elem = elemCount.first;
                    }
                }
            }
            return elem;
        }
    };

File: Typescript/majorityElement.ts
function majorityElement(nums: number[]): number {
    const elemCountMap: { [key: number]: number } = {};

    for (const num of nums) {
        if (elemCountMap[num] === undefined) {
            elemCountMap[num] = 1;
        } else {
            elemCountMap[num]++;
        }
    }

    let maxCount = 0;
    let majorityElem = -1;

    for (const num in elemCountMap) {
        if (elemCountMap.hasOwnProperty(num)) {
            const count = elemCountMap[num];
            if (count > nums.length / 2 && count > maxCount) {
                maxCount = count;
                majorityElem = Number(num);
            }
        }
    }

    return majorityElem;
}